src/module.c

```diff
@@ -1099,10 +1103,15 @@ int RM_CreateSubcommand(RedisModuleCommand *parent, const char *name, RedisModul
         return REDISMODULE_ERR; /* A parent command should be a pure container of subcommands */
 
     /* Check if the command name is busy within the parent command. */
-    if (parent_cmd->subcommands_dict && lookupCommandByCStringLogic(parent_cmd->subcommands_dict, name) != NULL)
+    sds sub_name = sdsnew(name);
+    if (parent_cmd->subcommands_dict && lookupSubcommand(parent_cmd, sub_name) != NULL) {
```

Comment:
first, create the fullname by calling `catSubCommandFullname` and then lookup in the parent_cmd

the fullname will later be sent to moduleCreateCommandProxy

Reply:
The hash function `dictSdsCaseHash` use the `sdslen`. maybe we still need the `sds sub_name`?
```
dictEntry *dictFind(dict *d, const void *key)
{
    dictEntry *he;
    uint64_t h, idx, table;

    if (dictSize(d) == 0) return NULL; /* dict is empty */
    if (dictIsRehashing(d)) _dictRehashStep(d);
    h = dictHashKey(d, key);

uint64_t dictSdsCaseHash(const void *key) {
    return dictGenCaseHashFunction((unsigned char*)key, sdslen((char*)key));
}
```

Reply:
yes, you're right

so you can ignore this comment

Reply:
good idea!

so:
1.`sds fullanme` is always and sds and it contains the fullname
2. `const char *declared_name` is a `const char *` for native commands and `sds` for module commands. it contains the sub_name (which is just the full name for non-subcommands)

and the signature is:
```
// takes over declared_name and fullname
RedisModuleCommand *moduleCreateCommandProxy(struct RedisModule *module, sds declared_name, sds fullname, RedisModuleCmdFunc cmdfunc, int64_t flags, int firstkey, int lastkey, int keystep)
```
right?

Reply:
yes